### ISSUE TITLE
[Feature Request] Automatic Default Layer Switching Based on Bluetooth Profile in ZMK

### ISSUE THREAD

**Issue by ozkan** [zmkfirmware/zmk#2246]
**Description:**
As a user of ZMK firmware for custom keyboards, I often switch between different devices with varying Bluetooth profiles. Currently, switching between Bluetooth profiles manually is seamless with ZMK. However, it would greatly enhance the usability of ZMK firmware if it could automatically switch the default layer based on the selected Bluetooth profile.

**Proposal:**
Implement a feature in ZMK firmware that automatically switches the default layer based on the selected Bluetooth profile. This would allow users to have different key mappings tailored to each Bluetooth profile, eliminating the need for manual layer switching each time the Bluetooth profile changes.

**Benefits:**

- Improved user experience: Automatic default layer switching would streamline the keyboard setup process for users who frequently switch between devices.

- Increased versatility: Users would be able to customize key mappings for different Bluetooth profiles, enhancing the flexibility of ZMK firmware.

- Simplified workflow: Eliminating the need for manual layer switching saves time and effort for users.

**Comment by elpekenin** [comment]
I plan to PR that soon, see #2222 for some info.

In the meantime, you can use (and give feedback, please) the intial implementation i have done  (https://github.com/elpekenin/zmk-userspace)

**Comment by and-elf** [comment]
Wouldn't it easier to use a macro that chooses BT profile and selects a specific layer?

**Comment by fitzage** [comment]
@and-elf Your comment was exactly the kick in the pants I needed to set up the macros. Perfect. No extra features required.

**Comment by repcakk** [comment]
Is there any workaround for this functionality like liking BT profiles with dedicated layers?

**Comment by snoyer** [comment]
There's been some attempts such as #2222 mentioned above and #1500 but neither have gotten significant traction so far.
@nmunnich mentioned that a proper solution could be developed on top of #2717 which wasn't ready at the time but has been merged since so maybe they can give us an update on the way forward

**Comment by nmunnich** [comment]
Yeah I think we could do something nice now. I did try doing something in this direction, but then I ended up getting distracted with a larger refactoring of the related parts of the codebase. What I would envision is something along the lines of
- "layers-active-on-boot" being a dts property on the keymap, taking in a list that gets turned into a bitmask similar to how combos do it
- "save-layers-to-endpoint" allowing the saving of currently locked-on layers to the currently active endpoint

iirc when I played around with it, the difference in treatment between usb as an endpoint and different BLE profiles as endpoints is what ended up sidetracking me. I'll get around to doing something eventually if nobody else does before me, but I can't make any guarantees or even hypotheticals about timelines if someone else does end up tackling it.
